workitem/workitem_repository_blackbox_test.go

```diff
@@ -108,16 +108,32 @@ func (s *workItemRepoBlackBoxTest) TestSave() {
 	})
 
 	s.T().Run("change is not prohibited", func(t *testing.T) {
-		// tests that you can change the type of a work item. NOTE: This
-		// functionality only works on the DB layer and is not exposed to REST.
 		// given
 		fxt := tf.NewTestFixture(t, s.DB, tf.WorkItems(1), tf.WorkItemTypes(2))
 		// when
 		fxt.WorkItems[0].Type = fxt.WorkItemTypes[1].ID
 		newWi, err := s.repo.Save(s.Ctx, fxt.WorkItems[0].SpaceID, *fxt.WorkItems[0], fxt.Identities[0].ID)
 		// then
-		require.NoError(s.T(), err)
-		assert.Equal(s.T(), fxt.WorkItemTypes[1].ID, newWi.Type)
+		require.NoError(t, err)
+		assert.Equal(t, fxt.WorkItemTypes[1].ID, newWi.Type)
+	})
+
+	s.T().Run("change of type along with field is not prohibited", func(t *testing.T) {
```

Comment:
I think we need a test for this on repo level as well where this should actually be allowed as I've described above.